tests/foreman/cli/test_realm.py

```diff
@@ -0,0 +1,316 @@
+                Realm.info({'id': realm['id']})
+
+    @tier1
+    def test_positive_create_delete_id(self):
+        """Proxy deletion by realm id
+
+        :id: 7c1aca0e-9724-40de-b38f-9189bdae0514
+
+        :expectedresults: Proxy is created and deleted
+        """
+        name = gen_string('alpha', random.randint(1, 30))
+        realm_name = gen_string('alpha', random.randint(1, 30))
+        with self.subTest(name):
+            proxy = self._make_proxy({u'name': name})
+            realm = make_realm({
+                'name': realm_name,
+                'realm-proxy-id': proxy['id'],
+                'realm-type': 'Red Hat Identity Management',
+                'locations': proxy['locations']
+            })
+            self.assertEqual(realm['name'], realm_name)
+            Realm.delete({'id': realm['id']})
+            with self.assertRaises(CLIReturnCodeError):
+                Realm.info({'id': realm['id']})
+
+    @tier1
+    def test_positive_realm_info_name(self):
+        """Test realm info functionality
+
+        :id: 2e3e92df-61f3-4c6b-98b9-dc9c2f8d140c
+
+        :expectedresults: Realm info information from name is correct
+        """
+        name = gen_string('alpha', random.randint(1, 30))
+        with self.subTest(name):
+            proxy = self._make_proxy({u'name': name})
+            realm = make_realm({
+                'name': 'info_test',
+                'realm-proxy-id': proxy['id'],
+                'realm-type': 'Red Hat Identity Management',
+                'locations': proxy['locations']
+            })
+            info = Realm.info({'name': realm['name']})
+            for key in info.keys():
+                self.assertEquals(info[key], realm[key])
+            Realm.delete({'name': realm['name']})
+
+    @tier1
+    def test_positive_realm_info_id(self):
+        """Test realm info functionality
+
+        :id: 1ae7b3af-221e-4480-9e93-d05d573456b4
+
+        :expectedresults: Realm info information from ID is correct
+        """
+        name = gen_string('alpha', random.randint(1, 30))
+        with self.subTest(name):
+            proxy = self._make_proxy({u'name': name})
+            realm = make_realm({
+                'name': 'info_test',
+                'realm-proxy-id': proxy['id'],
+                'realm-type': 'Red Hat Identity Management',
+                'locations': proxy['locations']
+            })
+            info = Realm.info({'id': realm['id']})
+            for key in info.keys():
+                self.assertEquals(info[key], realm[key])
+            self.assertEquals(info, Realm.info({'id': realm['id']}))
+            Realm.delete({'id': realm['id']})
+
+    @tier1
+    def test_positive_realm_update_name(self):
+        """Test updating realm name
+
+        :id: c09e6599-c77a-4290-ac93-311d06e3d860
+
+        :expectedresults: Realm name updated
+        """
+        name = gen_string('alpha', random.randint(1, 30))
```

Comment:
I suggest to have something like this:
```
realm_name = gen_string('alpha')
new_realm_name = gen_string('alphanumeric')
proxy = self._make_proxy({u'name': gen_string('alpha')})
realm = make_realm({
    'name': realm_name,
    'realm-proxy-id': proxy['id'],
    'realm-type': 'Red Hat Identity Management',
    'locations': proxy['locations']
})
self.assertEquals(realm['name'], realm_name)
Realm.update({'id': realm['id'], 'new-name': new_realm_name})
realm = Realm.info({'id': realm['id']})
self.assertEquals(realm['name'], new_realm_name)
Realm.delete({'id': realm['id']})
```
- I don't think that we need to use subTest for every test as proxy is not our primary goal here
- We try to omit numbers in variable names
- It is better to have dynamic names
- We try to omit checking result output message in case we have alternative to have better support of future localization ideology